compare_experiments/nbconverted/compare_generic_genes.py

```diff
@@ -0,0 +1,210 @@
+
+
+# In[12]:
+
+
+# Distribution of concordant genes in template experiment 2
+template2_mean = template_2.mean()
+print("Percent concordant genes with 0 expression in template 2:",
+      len(template2_mean[concordant_genes].loc[template2_mean[concordant_genes]==0])/len(template2_mean[concordant_genes]))
+
+print("Percent nonzero concordant genes in template 2:",
+      len(template2_mean[concordant_genes].loc[(template2_mean[concordant_genes]>0) &
+                                                (template2_mean[concordant_genes]<1000)])/len(template2_mean[concordant_genes]))
+
+# There are more 0 expressed genes in this template experiment
+sns.distplot(template_2.mean()[concordant_genes], kde=False)
+
+
+# In[13]:
+
+
+# Distribution of discordant gense in template experiment 1
+template1_mean = template_1.mean()
+
+print("Percent discordant genes with 0 expression in template 1:", 
+      len(template1_mean[discordant_genes].loc[template1_mean[discordant_genes]==0])/len(template1_mean[discordant_genes]))
+
+print("Percent nonzero discordant genes in template 1:", 
+      len(template1_mean[discordant_genes].loc[(template1_mean[discordant_genes]>0) &
+                                           (template1_mean[discordant_genes]<1000)])/len(template1_mean[discordant_genes]))
+
+print(len(template1_mean[discordant_genes].loc[template1_mean[discordant_genes]>0])/len(template1_mean[discordant_genes]))
+sns.distplot(template_1.mean()[discordant_genes], kde=False)
+
+
+# In[14]:
+
+
+# Distribution of discordant genes in template experiment 2
+template2_mean = template_2.mean()
+
+print("Percent discordant genes with 0 expression in template 2:",
+      len(template2_mean[discordant_genes].loc[template2_mean[discordant_genes]==0])/len(template2_mean[discordant_genes]))
+
+print("Percent nonzero discordant genes in template 2:",
+      len(template2_mean[discordant_genes].loc[(template2_mean[discordant_genes]>0) &
+                                                (template2_mean[discordant_genes]<1000)])/len(template2_mean[discordant_genes]))
+
+sns.distplot(template_2.mean()[discordant_genes], kde=False)
+
+
+# **Takeaway:**
+# 
+# Doesn't appear to be much of a difference between the distribution of average gene expression values for these two experiments. 
+# 
+# Theoretically, I would expect the scenario where a gene is lowly expressed in the context of template experiment 1 and therefore not found to be generic. But this same gene could be found to be generic in the context of template experiment 2 if it is more expressed. Its possible that differences in gene expression distribution can change which genes are found to be generic given that the simulation is producing experiments with a similar context. 
+# 
+# In this case, despite having similar gene expression distributions there are still many differences in gene ranking. This suggests to me that level of gene expression activity doesn't matter as much as the overall patterns perhaps. 
```

Comment:
This makes sense looking at the distributions, doesn't it? If almost all genes have expression levels very close to 0, then small fluctuations in the expression of some genes could lead to large changes in rank without changing the overall (skewed) distribution very much.

It's definitely interesting to know that the overall distributions are the same (which is what you're showing here), but I think looking at a log plot (or even a log-log plot, since the x-axis has a pretty large range too) could help to elucidate some of the differences between lowly expressed genes that seem to be driving the changes in rank.

Reply:
This is a great point. Thank you for thinking about this. After scaling by logs, we observe a slight shift showing that concordant genes are more lowly expressed compared to discordant genes, but most genes are still predominantly lowly gene expression. This would follow what you're saying -- If most genes have expression levels very close to 0, then small fluctuations in the expression of some genes could lead to large changes in rank without changing the overall expression distribution.

I'll need to continue to think about this some more. But It would make sense to me that the patterns as opposed to the gene expression levels matter more for differentiating gene ranking. And that your logic for why similar distributions of expression can lead to differences in rank can explain why this is the case.